packages/admin-ui/src/Checkbox/CheckboxGroupPresenter.ts

```diff
@@ -0,0 +1,74 @@
+import { makeAutoObservable } from "mobx";
+import { CheckboxGroupVm } from "./CheckboxGroup";
+import { CheckboxItem } from "./CheckboxItem";
+import { CheckboxItemMapper } from "./CheckboxItemMapper";
+import { CheckboxItemDto } from "./CheckboxItemDto";
+
+interface CheckboxGroupPresenterParams<TValue = string | number> {
+    items: CheckboxItemDto[];
+    values: TValue[];
+    onCheckedChange: (values: TValue[]) => void;
+}
+
+interface ICheckboxGroupPresenter<TValue = string | number> {
+    vm: CheckboxGroupVm;
+    init: (params: CheckboxGroupPresenterParams<TValue>) => void;
+    changeChecked: (value: TValue) => void;
+}
+
+class CheckboxGroupPresenter<TValue = string | number> implements ICheckboxGroupPresenter<TValue> {
+    private params?: CheckboxGroupPresenterParams<TValue>;
+    private items: CheckboxItem[] = [];
+
+    constructor() {
+        this.params = undefined;
+        makeAutoObservable(this);
+    }
+
+    public init(params: CheckboxGroupPresenterParams<TValue>): void {
+        this.params = params;
+        this.items = this.getItems(params.items, params.values);
+    }
+
+    get vm(): CheckboxGroupVm {
+        return {
+            items: this.items.map(item => CheckboxItemMapper.toFormatted(item))
+        };
+    }
+
+    public changeChecked = (value: TValue) => {
+        if (!this.params?.items) {
+            return;
+        }
+
+        const currentValues = [...this.params.values];
+        const newValues = this.updateValues(currentValues, value);
+        this.items = this.getItems(this.params.items, newValues);
+        this.params.onCheckedChange(newValues);
+    };
+
+    private updateValues(currentValues: TValue[], value: TValue): TValue[] {
```

Comment:
Found the `updateValues` name a bit odd. 🤔 

It's a nitpick ofc, but renaming the fn to maybe `getUpdatedValues` clarifies the intent and makes it evident that the function doesn't mutate currentValues.

Reply:
👍